src/tiledb/cloud/file_ingestion/ingestion.py

```diff
@@ -0,0 +1,177 @@
+    return [
+        flattened_results[n * batch_size : (n + 1) * batch_size]
+        for n in range(num_chunks)
+    ]
+
+
+def ingest_files_udf(
+    dataset_uri: str,
+    file_uris: Sequence[str],
+    *,
+    namespace: Optional[str] = None,
+    verbose: bool = False,
+) -> None:
+    """
+    Ingest files.
+
+    :param dataset_uri: The dataset URI.
+    :param file_uris: An iterable of file URIs.
+    :param namespace: TileDB-Cloud namespace, defaults to None.
+    :param verbose: Verbose logging, defaults to False.
+    """
+    logger = get_logger_wrapper(verbose)
+
+    for file_uri in file_uris:
+        filename = file_uri.replace(",", "").replace(" ", "_")
+        filestore_array_uri = f"{dataset_uri}/{filename}"
+        filestore_array_s3_uri = f"s3{filestore_array_uri.split('s3')[1]}"
+        namespace = (
+            namespace or filestore_array_uri.split("tiledb://")[1].split("/s3")[0]
+        )
+        logger.debug(
+            f"""
+            ---------------------------------------------\n
+            - Filename: {filename}\n
+            - Namespace: {namespace}\n
+            - Array URI: {filestore_array_uri}\n
+            - S3 Array URI: {filestore_array_s3_uri}\n
+            ---------------------------------------------
+            """
+        )
+
+        if not tiledb.array_exists(filestore_array_uri):
+            tiledb.cloud.file.create_file(
+                namespace=namespace,
+                name=filename,
+                input_uri=file_uri,
+                output_uri=filestore_array_s3_uri,
+            )
+
+
+def ingest_files(
+    dataset_uri: str,
+    *,
+    acn: Optional[str] = None,
+    config: Optional[dict] = None,
+    namespace: Optional[str] = None,
+    source: Union[Sequence[str], str],
+    include: Optional[str] = None,
+    exclude: Optional[str] = None,
+    taskgraph_name: Optional[str] = DEFAULT_FILE_INGESTION_NAME,
+    batch_size: Optional[int] = DEFAULT_BATCH_SIZE,
+    resources: Optional[Mapping[str, Any]] = None,
+    verbose: bool = False,
+) -> tiledb.cloud.dag.DAG:
+    """
+    Ingest files into a dataset.
+
+    :param dataset_uri: The dataset URI
+    :param source: URI or an iterable of URIs of input files.
+    :param acn: Access Credentials Name (ACN) registered in TileDB Cloud (ARN type),
+        defaults to None
+    :param config: Config dictionary, defaults to None
+    :param namespace: TileDB-Cloud namespace, defaults to None
+    :param include: UNIX shell style pattern to filter files in the search,
+        defaults to None
+    :param exclude: UNIX shell style pattern to filter files out of the search,
+        defaults to None
+    :param taskgraph_name: Optional name for taskgraph, defaults to "file-ingestion"
+    :param batch_size: Batch size for file ingestion, defaults to 100
+    :param resources: Configuration for node specs,
+        defaults to {"cpu": "2", "memory": "8Gi"}
```

Comment:
I think we can do smaller resources. Since you aren't ingesting the file client side this is really just a listing operation.